core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -187,6 +188,7 @@ public synchronized void connect() throws AlluxioStatusException {
       LOG.info("Alluxio client (version {}) is trying to connect with {} @ {}",
           RuntimeConstants.VERSION, getServiceName(), mAddress);
 
+      assert mAddress != null : "@AssumeAssertion(nullness)";
```

Comment:
@aaudiber In the following case ```mAddress``` is initialized to null:
https://github.com/Alluxio/alluxio/blob/master/core/common/src/main/java/alluxio/AbstractMasterClient.java#L36

Thus, the checker framework required to make ```mAddress``` variable ```@Nullable```, this can be solved by having another constructor perhaps with a default value for ```mAddress``` what are your thoughts?

At the moment it's actually possible that ```getAddress()``` return a null value, if we extend from ```AbstractClient```, but don't overwrite the method ```getAddress()```.